src/Games/NexusMods.Games.StardewValley/Installers/GenericInstaller.cs

```diff
@@ -0,0 +1,118 @@
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using DynamicData.Kernel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.IO;
+using NexusMods.Abstractions.Library.Installers;
+using NexusMods.Abstractions.Library.Models;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Extensions.BCL;
+using NexusMods.Games.StardewValley.Models;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using SMAPIManifest = StardewModdingAPI.Toolkit.Serialization.Models.Manifest;
+
+namespace NexusMods.Games.StardewValley.Installers;
+
+/// <summary>
+/// Generic installer for Stardew Valley that installs files into the "Mods" directory.
+/// </summary>
+public class GenericInstaller : ALibraryArchiveInstaller
+{
+    private readonly IFileStore _fileStore;
+
+    public GenericInstaller(IServiceProvider serviceProvider, ILogger<GenericInstaller> logger) : base(serviceProvider, logger)
+    {
+        _fileStore = serviceProvider.GetRequiredService<IFileStore>();
+    }
+
+    public override async ValueTask<InstallerResult> ExecuteAsync(
+        LibraryArchive.ReadOnly libraryArchive,
+        LoadoutItemGroup.New loadoutGroup,
+        ITransaction transaction,
+        Loadout.ReadOnly loadout,
+        CancellationToken cancellationToken)
+    {
+        var manifestFiles = await GetManifestsAsync(libraryArchive, cancellationToken).ToArrayAsync(cancellationToken: cancellationToken);
+        var manifestFileIds = manifestFiles.Select(static x => x.FileEntry.Id).ToHashSet();
+
+        var previousInMods = Optional<bool>.None;
+
+        foreach (var fileEntry in libraryArchive.Children)
+        {
+            var path = fileEntry.Path;
+            if (path.InFolder(Constants.ContentFolder)) return new NotSupported();
+
+            var inModsFolder = path.InFolder(Constants.ModsFolder);
+            if (previousInMods.HasValue)
+            {
+                // NOTE(erri120): this installer doesn't support this case where a mod is
+                // packages to be installed into the root of the game folder
+                // /foo
+                // /Mods/bar
+                if (previousInMods.Value != inModsFolder) return new NotSupported();
+            }
+            else
+            {
+                previousInMods = inModsFolder;
+            }
```

Comment:
Wouldn't this break if the library archive contained something like:
```
- LookupAnything
- - Manifest.json
- - [OtherStuff]
```

Reply:
What this guards against is a mix of files in `Mods` and not in `Mods`.